esphome/components/wifi/wifi_component.cpp

```diff
@@ -35,6 +35,7 @@ float WiFiComponent::get_setup_priority() const { return setup_priority::WIFI; }
 
 void WiFiComponent::setup() {
   ESP_LOGCONFIG(TAG, "Setting up WiFi...");
+  this->dump_config();  // Print config before connecting
```

Comment:
The dump config is really only useful once it's already connected, let's just log the MAC only...

```suggestion
  ESP_LOGCONFIG(TAG, "  Local MAC: %s", get_mac_address_pretty().c_str());
```